Source/Events.Processing/Filters/FiltersService.cs

```diff
@@ -123,6 +128,134 @@ public class FiltersService : FiltersBase
+                {
+                    _logger.Warning(failure.Reason);
+                    await PartitionedWriteFailedRegistrationResponse(dispatcher, failure, context.CancellationToken).ConfigureAwait(false);
+                    return;
+                }
+
+                await dispatcher.Accept(new FilterRegistrationResponse(), context.CancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                registrationResults
+                    .SelectMany(tenantAndResult => new[] { tenantAndResult.Item2.FilterStreamProcessor })
+                    .ForEach(_ => _.Dispose());
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async Task ConnectPublic(
+            IAsyncStreamReader<PublicFiltersClientToRuntimeMessage> runtimeStream,
+            IServerStreamWriter<FilterRuntimeToClientMessage> clientStream,
+            ServerCallContext context)
+        {
+            var dispatcher = _reverseCallDispatchers.GetFor<PublicFiltersClientToRuntimeMessage, FilterRuntimeToClientMessage, PublicFiltersRegistrationRequest, FilterRegistrationResponse, FilterEventRequest, PartitionedFilterResponse>(
+                runtimeStream,
+                clientStream,
+                context,
+                _ => _.RegistrationRequest,
+                (serverMessage, registrationResponse) => serverMessage.RegistrationResponse = registrationResponse,
+                (serverMessage, request) => serverMessage.FilterRequest = request,
+                _ => _.FilterResult,
+                _ => _.CallContext,
+                (request, context) => request.CallContext = context,
+                _ => _.CallContext);
+
+            if (!await dispatcher.ReceiveArguments(context.CancellationToken).ConfigureAwait(false))
+            {
+                const string message = "Public Filters connection arguments were not received";
+                _logger.Warning(message);
+                var failure = new Failure(FiltersFailures.NoFilterRegistrationReceived, message);
+                await PublicWriteFailedRegistrationResponse(dispatcher, failure, context.CancellationToken).ConfigureAwait(false);
+                return;
+            }
+
+            var arguments = dispatcher.Arguments;
+            var filterId = arguments.FilterId.To<StreamId>();
+            if (filterId.IsNonWriteable)
+            {
+                _logger.Warning("Received public filter registration request with Filter Id: '{filterId}' which is an invalid stream id", filterId);
+                var failure = new Failure(FiltersFailures.CannotRegisterFilterOnNonWriteableStream, $"Received public filter registration request with Filter Id: '{filterId}' which is an invalid stream id");
+                await PublicWriteFailedRegistrationResponse(dispatcher, failure, context.CancellationToken).ConfigureAwait(false);
+                return;
+            }
```

Comment:
This logic is duplicated multiple times - just with different response types. So that could be pulled out to a generic method. But that might be one of those things that you were thinking of improving later either way.

Reply:
Resolved by later PR